### DISCUSSION TITLE
DataGrid show only RowDetails

### DISCUSSION THREAD

**Discussion by GundlackFlorian** [AvaloniaUI/Avalonia#11197] [Q&A] [answered]
Hello,

I am trying to work on a way to sort a ListBox like a DataGrid.
The most easiest way I found, was to simply:

- use a DataGrid
- put my Content in the RowDetailsTemplate and set it to always visible
- Hide the DataGridCellsPresenter

I also added a Checkbox in the DataGrid Header to switch between custom mode and regular grid mode.

This kinda works, the problem is that sometimes some rows are not shown, and if I drag the scrollbar the rows are not shown at all (and some other issues).
I tried some stuff like invalidating visuals, but with no success, so I am a little bit stuck.

I made a very simple repo, with the CustomGrid and some Faker data, it's on 11-Preview7 (because of the usage of `ControlTheme`).
https://github.com/Gundz/CustomDataGrid

Thanks in advance for your help.

Ps: it could be nice to have this as a feature, by adding a "Only" option to the `RowDetailsVisibilityMode` for example,
I will try but not sure I'm competent enough.

**Comment by timunie** [comment]
Hi @Gundz, 

wouldn't be a Button (can be styled like a DataGridHeader) enough? The idea would be to bind to a command in the ViewModel. The Filtering and sorting can be done by using `DynmaicData`. 

Happy coding
Tim

**Reply by GundlackFlorian** [reply]
Hello,

I first started looking at this solution, but I have to make a System to display and bind all the column members, style it, add the littles sort icons and so on.
And it's just a "lot" of work to just reinvent the wheel, everything is already there in the DataGrid.

**Comment by GundlackFlorian** [comment, marked as answer]
Hello,

Little update, I got back on this project,
It seems that it was a bug, the fact that the rows where not shown was an issue with DataGrid not always passing the DataContext of the Item properly.
After upgrading avalonia to a more recent version it's fixed and works fine.
